bot.ml

```diff
@@ -786,15 +786,158 @@ let callback _conn req body =
+                ~number:pr.issue.number
+              >>= function
+              | Ok reviews_info -> (
+                match
+                  List.find comment_info.issue.labels ~f:(fun label ->
+                      string_match ~regexp:"needs:.*" label)
+                with
+                | Some l ->
+                    GitHub_mutations.post_comment ~bot_info
+                      ~message:
+                        (f "@%s: Needs label %s found, merge aborted."
+                           comment_info.author l)
+                      ~id:pr.id
+                | None -> (
+                    if
+                      not
+                        (List.exists comment_info.issue.labels ~f:(fun label ->
+                             string_match ~regexp:"kind:.*" label))
+                    then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No kind label found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else if not comment_info.issue.milestoned then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No milestone found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else
+                      match reviews_info.review_decision with
+                      | REVIEW_REQUIRED | CHANGES_REQUESTED ->
+                          GitHub_mutations.post_comment ~bot_info
+                            ~message:
+                              (f
+                                 "@%s: PR cannot be merged as it isn't \
+                                  approved yet."
+                                 comment_info.author)
+                            ~id:pr.id
+                      | APPROVED -> (
+                          if
+                            not
+                              (List.exists comment_info.issue.assignees
+                                 ~f:(fun login ->
+                                   String.equal login comment_info.author))
+                          then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR as you're not \
+                                    among the assignees."
+                                   comment_info.author)
+                              ~id:pr.id
+                          else if String.equal comment_info.author pr.user then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR since you \
+                                    authored it."
```

Comment:
```suggestion
                                   "@%s: You can't merge the PR because you \
                                    are the author."
```